lib/HttpClient.ts

```diff
@@ -177,20 +177,26 @@ export class HttpClient {
     public async request(verb: string, requestUrl: string, data: string | NodeJS.ReadableStream, headers: ifm.IHeaders): Promise<HttpClientResponse> {
         let info: RequestInfo = this._prepareRequest(verb, requestUrl, headers);
         let response: HttpClientResponse = await this._requestRaw(info, data);
-        
+
         let redirectsRemaining: number = this._maxRedirects;
         while (HttpRedirectCodes.indexOf(response.message.statusCode) != -1
                && this._allowRedirects
                && redirectsRemaining > 0) {
-          const location: any = response.message.headers["location"];
 
-          if (!location) {
-             throw new Error(`Unable to find location header after HTTP ${response.message.statusCode} redirect.`);
-          }
+            const redirectUrl: any = response.message.headers["location"];
+            if (!redirectUrl) {
+                // if there's no location to redirect to, we won't
+                break;
+            }
+
+            // we need to finish reading the response before reassigning response
+            // which will leak the open socket.
+            response.readBody();
```

Comment:
is this synchronous or are you missing an await here? otherwise looks fine to me

Reply:
Ahhh - great catch 🥇  .  It worked because it probably only needed to read one block off the stream and it wasn't the last test in the sample